src/components/application_manager/test/application_manager_impl_test.cc

```diff
@@ -195,6 +212,417 @@ TEST_F(
+  converted_params["width"] = 480;
+
+  std::vector<std::string> empty;
+
+  // check: SetVideoConfig() and StartStreaming() are called
+  EXPECT_CALL(*mock_app_ptr_, SetVideoConfig(service_type, converted_params))
+      .WillOnce(DoAll(InvokeMemberFuncWithArg4(
+                          app_manager_impl_.get(),
+                          &ApplicationManagerImpl::OnStreamingConfigured,
+                          session_key,
+                          service_type,
+                          true,
+                          ByRef(empty)),
+                      Return(true)));
+  EXPECT_CALL(*mock_app_ptr_, StartStreaming(service_type)).WillOnce(Return());
+
+  app_manager_impl_->OnServiceStartedCallback(
+      device_handle, session_key, service_type, &input_params);
+
+  // check: return value is true and list is empty
+  EXPECT_TRUE(result);
+  EXPECT_TRUE(rejected_params.empty());
+}
+
+static bool ValidateList(std::vector<std::string>& expected,
+                         std::vector<std::string>& actual) {
+  if (expected.size() != actual.size()) {
+    return false;
+  }
+  for (unsigned int i = 0; i < expected.size(); i++) {
+    std::string& param = expected[i];
+    unsigned int j;
+    for (j = 0; j < actual.size(); j++) {
+      if (param == actual[j]) {
+        break;
+      }
+    }
+    if (j == actual.size()) {
+      // not found
+      return false;
+    }
+  }
+  return true;
+}
+
+TEST_F(ApplicationManagerImplTest,
+       OnServiceStartedCallback_VideoSetConfig_FAILURE) {
+  AddMockApplication();
+
+  const connection_handler::DeviceHandle device_handle = 0;
+  const protocol_handler::ServiceType service_type =
+      protocol_handler::ServiceType::kMobileNav;
+  const int32_t session_key = 123;
+  EXPECT_CALL(*mock_app_ptr_, app_id()).WillRepeatedly(Return(session_key));
+  EXPECT_CALL(*mock_app_ptr_, is_navi()).WillRepeatedly(Return(true));
+  EXPECT_CALL(*mock_app_ptr_, hmi_level())
+      .WillRepeatedly(Return(mobile_apis::HMILevel::HMI_FULL));
+
+  bool result = false;
+  std::vector<std::string> rejected_params;
+  EXPECT_CALL(mock_connection_handler_, NotifyServiceStartedResult(_, _))
+      .WillOnce(DoAll(SaveArg<0>(&result), SaveArg<1>(&rejected_params)));
+
+  BsonObject input_params;
+  bson_object_initialize_default(&input_params);
+  char protocol_version[] = "1.0.0";
+  bson_object_put_string(&input_params, "protocolVersion", protocol_version);
+  bson_object_put_int64(&input_params, "mtu", 100);
+  char protocol_name[] = "RTP";
+  bson_object_put_string(&input_params, "videoProtocol", protocol_name);
+  char codec_name[] = "VP9";
+  bson_object_put_string(&input_params, "videoCodec", codec_name);
+  bson_object_put_int32(&input_params, "desiredHeight", 640);
+  bson_object_put_int32(&input_params, "desiredWidth", 480);
+
+  smart_objects::SmartObject converted_params(smart_objects::SmartType_Map);
+  converted_params["protocol"] = hmi_apis::Common_VideoStreamingProtocol::RTP;
```

Comment:
Done in https://github.com/smartdevicelink/sdl_core/pull/1700/commits/229a2901e652be32d39e5d8559fb05f1a794dde5.